cgroup2/utils.go

```diff
@@ -379,41 +389,55 @@ func systemdUnitFromPath(path string) string {
 func readHugeTlbStats(path string) []*stats.HugeTlbStat {
 	usage := []*stats.HugeTlbStat{}
 	keyUsage := make(map[string]*stats.HugeTlbStat)
+
 	f, err := os.Open(path)
 	if err != nil {
 		return usage
 	}
-	files, err := f.Readdir(-1)
+
+	files, err := f.Readdirnames(-1)
 	f.Close()
 	if err != nil {
 		return usage
 	}
 
+	buf := make([]byte, 32)
 	for _, file := range files {
-		if strings.Contains(file.Name(), "hugetlb") &&
-			(strings.HasSuffix(file.Name(), "max") || strings.HasSuffix(file.Name(), "current")) {
-			var hugeTlb *stats.HugeTlbStat
-			var ok bool
-			fileName := strings.Split(file.Name(), ".")
+		if strings.Contains(file, "hugetlb") &&
+			(strings.HasSuffix(file, "max") || strings.HasSuffix(file, "current")) {
+			var (
+				hugeTlb *stats.HugeTlbStat
+				ok      bool
+			)
+			fileName := strings.Split(file, ".")
 			pageSize := fileName[1]
 			if hugeTlb, ok = keyUsage[pageSize]; !ok {
 				hugeTlb = &stats.HugeTlbStat{}
 			}
 			hugeTlb.Pagesize = pageSize
-			out, err := os.ReadFile(filepath.Join(path, file.Name()))
+
+			f, err := os.Open(filepath.Join(path, file))
 			if err != nil {
 				continue
 			}
+			defer f.Close()
+
+			n, err := f.Read(buf)
+			if err != nil {
+				continue
+			}
+
 			var value uint64
-			stringVal := strings.TrimSpace(string(out))
+			stringVal := strings.TrimSpace(string(buf[:n]))
```

Comment:
Another realization: if we use the bytes package; its again faster; does not change allocs though.
See: https://go.dev/play/p/7jPtYNfN7xq

I get 
```
Running tool: /usr/local/go/bin/go test -benchmem -run=^$ -bench ^(BenchmarkByteCompare|BenchmarkStringCompare)$ github.com/manugupt1/file -count=1 -timeout 30m

goos: linux
goarch: amd64
pkg: github.com/manugupt1/file
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkByteCompare-8     	64981508	        17.42 ns/op	       0 B/op	       0 allocs/op
BenchmarkStringCompare-8   	41275358	        27.36 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/manugupt1/file	3.223s
```

Reply:
I will do these for cgroup1 if you are fine with it.

Reply:
Go for it! That's a good point, we can defer the converting to strings after we trim